Source/DafnyDriver/DafnyDriver.cs

```diff
@@ -734,12 +621,13 @@ private record TargetPaths(string Directory, string Filename) {
       Contract.Assert(hasMain == (callToMain != null));
       bool targetProgramHasErrors = dafnyProgram.reporter.Count(ErrorLevel.Error) != oldErrorCount;
 
-      compiler.Coverage.WriteLegendFile();
+      compiler.WriteCoverageLegendFile();
 
       // blurt out the code to a file, if requested, or if other target-language files were specified on the command line.
-      var paths = GenerateTargetPaths(compiler, dafnyProgramName);
+      var paths = GenerateTargetPaths(dafnyProgramName);
       if (DafnyOptions.O.SpillTargetCode > 0 || otherFileNames.Count > 0 || (invokeCompiler && !compiler.SupportsInMemoryCompilation) ||
           (invokeCompiler && compiler.TextualTargetIsExecutable && !DafnyOptions.O.RunAfterCompile)) {
+        compiler.CleanSourceDirectory(paths.SourceDirectory);
```

Comment:
Do we always want to do that? I'm personally relying on the generated files to debug any problem.

Reply:
oh great, I thought it was cleaning at the end.